File: study/22.03.29/HJH/로또의 최고 순위와 최저 순위.java
import java.util.*;
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        // Arrays.sort(lottos);
        // Arrays.sort(win_nums);
        int zeros = 0;
        int fits = 0;
        int min = 0;
        int max = 0;
        for (int i = 0; i < 6; i++) {
            if (lottos[i] == 0) {
                zeros++;
                continue;
            } else {
                for (int j = 0; j < 6; j++) {
                    if(lottos[i] == win_nums[j]) {
                        fits++;
                        break;
                    }
                }
            }
        }
        min = rank(fits);
        max = rank(fits + zeros);
        int[] answer = {max, min};
        return answer;
    }
    
    public static int rank(int fits) {
        switch (fits) {
            case 6: return 1;
            case 5: return 2;
            case 4: return 3;
            case 3: return 4;
            case 2: return 5;
        }
        return 6;
    }
}

File: study/2023/23.02.07/HJH/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    let cntZeros = 0;
    let cntFits = 0;
  
    for (let n of lottos) {
      if (n === 0) cntZeros++;
      else if (win_nums.includes(n)) cntFits++;
    }
  
    const rank = 7 - (cntFits === 0 ? 1 : cntFits);
    const lowRank = cntZeros === 6 ? 5 : cntZeros;
    
    return [rank - lowRank, rank];
    // return [7 - (cntFits === 0 ? 1 : cntFits) - (cntZeros === 6 ? 5 : cntZeros), 7 - (cntFits === 0 ? 1 : cntFits)];
  }
